src/arch/riscv/isa.cc

```diff
@@ -331,6 +334,60 @@ ISA::setMiscReg(int misc_reg, RegVal val)
         warn("Ignoring write to %s.\n", CSRData.at(misc_reg).name);
     } else {
         switch (misc_reg) {
+
+          case MISCREG_PMPCFG0:
+          case MISCREG_PMPCFG2:
+            {
+
+                // might need to assert that we are
+                // in the M mode.
+
+                // PMP registers should only be modified in M mode
+                assert(readMiscRegNoEffect(MISCREG_PRV) == PRV_M);
+
+                // Do the specs define some sequencing!
+                // At least that's what qemu is doing!
+
+                std::cout << "PMP cfg update " << std::endl;
+
+                int i;
+                for (i=0; i < sizeof(val); i++)
+                {
+
+                uint8_t cfg_val = (val >> 8 * i)  & 0xff;
+                auto mmu = dynamic_cast<RiscvISA::MMU *>(tc->getMMUPtr());
+
+                auto pp = dynamic_cast<PMP *>(mmu->getPMP());
+
+                pp->pmp_update_cfg(i+(4*(misc_reg-MISCREG_PMPCFG0)),
+                                  cfg_val);
+                }
+                setMiscRegNoEffect(misc_reg, val);
```

Comment:
Odd whitespace. `{` go on the same line and the for loop

Reply:
Fixed.